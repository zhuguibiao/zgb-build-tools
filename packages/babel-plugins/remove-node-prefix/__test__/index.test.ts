// babel-plugin-demo/tests/test.js
import { describe, it, expect } from "vitest";
import { transformSync } from "@babel/core";
import myBabelPlugin from "../index";

describe("my-babel-plugin", () => {
  it("should replace 'hello' with 'world'", () => {
    const inputCode = `
        import fs from 'node:fs';
        import path from 'node:path';

        const dynamic = import('node:url');
        const fs2 = require('node:fs');
        const path2 = require.resolve('node:path');

        export * as fs from 'node:fs';
        export * from 'node:crypto';
        export { readFile } from 'node:fs';

        module.exports = { fs: require('node:fs') };
        module.exports.fs = require('node:fs');

        /** ignore */
        console.log('node:fs');
        fn('node:fs');
    `
    const output = transformSync(inputCode, { plugins: [myBabelPlugin] });
    console.log('输出:', output.code)
    expect(output.code).toMatchSnapshot();
  });
});

