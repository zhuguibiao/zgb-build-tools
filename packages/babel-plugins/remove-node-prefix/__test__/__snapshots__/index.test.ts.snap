// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`my-babel-plugin > should replace 'hello' with 'world' 1`] = `
"import fs from "fs";
import path from "path";
const dynamic = import("url");
const fs2 = require("fs");
const path2 = require.resolve("path");
export * as fs from "fs";
export * from "crypto";
export { readFile } from "fs";
module.exports = {
  fs: require("fs")
};
module.exports.fs = require("fs");

/** ignore */
console.log('node:fs');
fn('node:fs');"
`;
